include/command_line.h

```diff
@@ -1,229 +1,180 @@
-          throw std::invalid_argument{
-              "Encountered command line argument several times: " + argName};
+      if(pos != std::string::npos) {
+        auto argName = arg.substr(0, pos);
+        auto argVal = arg.substr(pos + 1);
+
+        if(args.find(argName) != args.end()) {
+          throw std::invalid_argument{"Encountered command line argument several times: " + argName};
         }
 
         args[argName] = argVal;
-      }
-      else
-      {
+      } else {
         flags.insert(arg);
       }
     }
   }
 
-  bool isArgSet(const std::string& arg) const
-  {
-    return args.find(arg) != args.end();
-  }
+  bool isArgSet(const std::string& arg) const { return args.find(arg) != args.end(); }
 
-  template<class T>
-  T getOrDefault(const std::string& arg, const T& defaultVal) const
-  {
+  template <class T>
+  T getOrDefault(const std::string& arg, const T& defaultVal) const {
     if(isArgSet(arg))
       return cast<T>(args.at(arg));
     return defaultVal;
   }
 
-  template<class T>
-  T get(const std::string& arg) const
-  {
-    try
-    {
+  template <class T>
+  T get(const std::string& arg) const {
+    try {
       return cast<T>(args.at(arg));
-    }
-    catch(std::out_of_range& e)
-    {
-      throw std::invalid_argument{"Command line argument was requested but missing: "+arg};
+    } catch(std::out_of_range& e) {
+      throw std::invalid_argument{"Command line argument was requested but missing: " + arg};
     }
   }
 
-  bool isFlagSet(const std::string& flag) const
-  {
-    return flags.find(flag) != flags.end();
-  }
+  bool isFlagSet(const std::string& flag) const { return flags.find(flag) != flags.end(); }
 
-  
 
 private:
-  
-
   CommandLineArguments args;
   FlagList flags;
 };
 
 
-struct VerificationSetting
-{
+struct VerificationSetting {
   bool enabled;
-  cl::sycl::id<3> begin = {0, 0, 0};
-  cl::sycl::range<3> range = {1, 1, 1};
+  sycl::id<3> begin = {0, 0, 0};
+  sycl::range<3> range = {1, 1, 1};
 };
 
-struct BenchmarkArgs
-{
+struct BenchmarkArgs {
   size_t problem_size;
   size_t local_size;
   size_t num_runs;
-  cl::sycl::queue device_queue;
+  sycl::queue device_queue;
   VerificationSetting verification;
   // can be used to query additional benchmark specific information from the command line
   CommandLine cli;
   std::shared_ptr<ResultConsumer> result_consumer;
 };
 
-class CUDASelector : public cl::sycl::device_selector {
-public:
-  int operator()(const cl::sycl::device& device) const override {
-    using namespace cl::sycl::info;
-    const std::string driverVersion = device.get_info<device::driver_version>();
-    if(device.is_gpu() && (driverVersion.find("CUDA") != std::string::npos)) {
-      return 1;
-    };
-    return -1;
-  }
-};
+int CUDASelector(const sycl::device& device) {
+  using namespace sycl::info;
+  const std::string driverVersion = device.get_info<device::driver_version>();
+  if(device.is_gpu() && (driverVersion.find("CUDA") != std::string::npos)) {
```

Comment:
Might this also select NVIDIA's OpenCL implementation? I think they have CUDA in the platform name or so.

Reply:
This is used just for DPC++ when targeting GPUs, but from what I know DPC++ can't target OpenCL on NVIDIA GPUs, am I right?
At this point, it would be better to have a device selector for each supported hardware, instead of relying on the device-specific one (e.g. cpu, gpu etc.)

Reply:
Ah, I see. I think DPC++ by default hides NVIDIA OpenCL devices because it does not support SPIR-V. Do we even need those device selectors? Both DPC++ and AdaptiveCpp support environment variables that can hide specific backends or even devices.

Reply:
Yeah I think the existing environment variables in addition to the device type selection in sycl bench should be enough